lib/logstash/filters/aggregate.rb

```diff
@@ -195,6 +195,28 @@ class LogStash::Filters::Aggregate < LogStash::Filters::Base
   # Example value : `"map['sql_duration'] += event['duration']"`
   config :code, :validate => :string, :required => true
 
+
+
+  # This configuration describes the code block that is executed either on timeout 
+  # or in case 'push_previous_map_as_event' is set to true.
+  #
+  # Example value : `"map['sql_duration'] += event['duration']"`
+  config :timeout_code, :validate => :string, :required => false
+
+
+  # This sets the event's key for the task id value. The aggregate plugin maps the aggregation map to 
+  # a task_id value. This task_id will then be set into the event. This can help correlate which events have been timed out
+  #
+  # This field has no default value and will not be set on the event if not configured.
+  #
+  # Example:
+  #
+  # If the taskId is "12345" and this field is set to "my_Id", the generated event will have:
+  # event[ "my_Id" ] = "12345"
+  #
+  config :timeout_task_id_field, :validate => :string, :required => false
+
```

Comment:
I have a slight update for this option description : 
```
This option indicates the timeout generated event's field for the "task_id" value. The task id will then be set into the timeout event. This can help correlate which tasks have been timed out.  
This field has no default value and will not be set on the timeout event if not configured.
Example:
If the task_id is "12345" and this field is set to "my_Id", the generated timeout event will have :
event[ "my_Id" ] = "12345"
```

I remove "The aggregate plugin maps the aggregation map to  task_id value. ", because explaining internal plugin behaviour is not necessary here, I think. "task_id" refers to the plugin option "task_id". So I think it is not necessary to explain more about that, and this could be confusing.